sigstore/_internal/rekor/client.py

```diff
@@ -82,114 +82,22 @@ def to_entry(self) -> RekorEntry:
             signed_entry_timestamp=self.signed_entry_timestamp,
         )
 
-
-@dataclass(frozen=True)
-class RekorEntry:
-    """
-    Represents a Rekor log entry.
-
-    Log entries are retrieved from Rekor after signing or verification events,
-    or generated from "offline" Rekor bundles supplied by the user.
-    """
-
-    uuid: Optional[str]
-    """
-    This entry's unique ID in the Rekor instance it was retrieved from.
-
-    For sharded Rekor deployments, IDs are unique per-shard.
-
-    Not present for `RekorEntry` instances loaded from offline bundles.
-    """
-
-    body: str
-    """
-    The base64-encoded body of the Rekor entry.
-    """
-
-    integrated_time: int
-    """
-    The UNIX time at which this entry was integrated into the Rekor log.
-    """
-
-    log_id: str
-    """
-    The log's ID (as the SHA256 hash of the DER-encoded public key for the log
-    at the time of entry inclusion).
-    """
-
-    log_index: int
-    """
-    The index of this entry within the log.
-    """
-
-    inclusion_proof: Optional[RekorInclusionProof]
-    """
-    An optional inclusion proof for this log entry.
-
-    Only present for entries retrieved from online logs.
-    """
-
-    signed_entry_timestamp: str
-    """
-    The base64-encoded Signed Entry Timestamp (SET) for this log entry.
-    """
-
     @classmethod
-    def from_response(cls, dict_: Dict[str, Any]) -> RekorEntry:
-        """
-        Create a new `RekorEntry` from the given API response.
-        """
-
-        # Assumes we only get one entry back
-        entries = list(dict_.items())
-        if len(entries) != 1:
-            raise RekorClientError("Received multiple entries in response")
-
-        uuid, entry = entries[0]
-
-        return cls(
-            uuid=uuid,
-            body=entry["body"],
-            integrated_time=entry["integratedTime"],
-            log_id=entry["logID"],
-            log_index=entry["logIndex"],
-            inclusion_proof=RekorInclusionProof.parse_obj(
-                entry["verification"]["inclusionProof"]
-            ),
-            signed_entry_timestamp=entry["verification"]["signedEntryTimestamp"],
-        )
-
-    def to_bundle(self) -> RekorBundle:
+    def from_entry(cls, entry: RekorEntry) -> RekorBundle:
```

Comment:
I moved this `to_bundle` method from the `RekorEntry` to a ``from_entry` class method. This minimises the public API and means that it doesn't have to know about this bundle type.

Reply:
That works for me. IMO it would also be acceptable to have the `RekorBundle` be part of the public API, since it's a well-defined public format already. But we can move forwards with this for now.